sources/metricdog/src/main.rs

```diff
@@ -0,0 +1,127 @@
+#![deny(rust_2018_idioms, unused_imports)]
+
+/*!
+# Introduction
+
+Metricdog sends anonymous information about the health of a Bottlerocket host.
+It does so by sending key-value pairs as query params in an HTTP GET request.
+
+Metricdog also has the ability to check that a list of critical services is running.
+It does so using `systemctl` and reports services that are not healthy.
+
+# What it Sends
+
+### The standard set of metrics:
+
+* `sender`: the application sending the report.
+* `event`: the event that invoked the report.
+* `version`: the Bottlerocket version.
+* `variant`: the Bottlerocket variant.
+* `arch`: the machine architecture, e.g.'x86_64' or 'arm'.
+* `aws_region`: the AWS region the machine is running in.
+* `seed`: the seed value used to roll-out updates.
+* `version_lock`: the optional setting that controls Bottlerocket update selection.
+* `ignore_waves`: an update setting that allows hosts to update before their seed is reached.
+
+### Additionally, when `metricdog` sends a 'health ping', it adds:
+
+* `is_healthy`: true or false based on whether critical services are running.
+* `failed_services`: a list of critical services that have failed, if any.
+
+# Configuration
+
+Configuration is read from a TOML file, which is generated from Bottlerocket settings:
+
+```toml
+# the url to which metricdog will send metrics information
+metrics_url = "https://example.com/metrics"
+# whether or not metricdog will send metrics. opt-out by setting this to false
+send_metrics = true
+# a list of systemd service names that will be checked
+service_checks = ["apiserver", "containerd", "kubelet"]
+# the AWS region
+aws_region = "us-west-2"
+# the update wave seed
+seed = 1234
+# what version bottlerocket should stay on
+version_lock = "latest"
+# whether bottlerocket should ignore update roll-out timing
+ignore_waves = false
+```
+*/
+
+#![deny(rust_2018_idioms)]
+
+mod args;
+mod config;
+mod error;
+#[cfg(test)]
+mod main_test;
+mod metricdog;
+#[cfg(test)]
+mod metricdog_test;
+mod service_check;
+
+use crate::args::{Arguments, Command};
+use crate::config::Config;
+use crate::error::Result;
+use crate::metricdog::Metricdog;
+use crate::service_check::{ServiceCheck, SystemdCheck};
+use bottlerocket_release::BottlerocketRelease;
+use log::error;
+use simplelog::{Config as LogConfig, SimpleLogger};
+use snafu::ResultExt;
+use std::process;
+use structopt::StructOpt;
+
+fn main() -> ! {
+    let args = Arguments::from_args();
+    SimpleLogger::init(args.log_level, LogConfig::default()).expect("unable to configure logger");
```

Comment:
What does `SimpleLogger` do for output?  Can we use `TermLogger` like we do elsewhere, so we can specify `TerminalMode::Mixed` to get errors on stderr?  I'm not sure the `StandardError` setting in `metricdog.service` will help us otherwise.

Reply:
Bottom-line: `SimpleLogger` does what we want (splits stderr/stdout) and doesn't require an interactive terminal to be attached.

`SimpleLogger` (does split between stderr/stdout)[https://github.com/Drakulix/simplelog.rs/blame/master/README.md#L13]. I have tested and confirmed this.

`TermLogger` (requires an interactive terminal)[https://github.com/Drakulix/simplelog.rs/blame/master/README.md#L14]. @etungsten can speak to some difficulties with the update operator because `updog` uses `TermLogger`. [This issue](https://github.com/bottlerocket-os/bottlerocket-update-operator/issues/29#issue-580275364) is a result of using `TermLogger` which required the update operator to attach a PTY terminal.

Reply:
That's a good find.  We don't need/want the color support from `TermLogger` everywhere, so we could use `SimpleLogger` most places.  That may solve #576 (@jahkeup) and #697 (@etungsten) too.

Reply:
Created https://github.com/bottlerocket-os/bottlerocket/issues/1157 to track SimpleLogger use.